internal/testutil/acc/shared_resource.go

```diff
@@ -9,12 +9,13 @@ import (
 )
 
 // SetupSharedResources must be called from TestMain test package in order to use ProjectIDExecution.
-func SetupSharedResources() {
+// It returns the cleanup function that must be called at the end of TestMain.
+func SetupSharedResources() func() {
```

Comment:
like this approach of having the setup and cleanup coupled together to avoid any potential strange use case